extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt

```diff
@@ -44,32 +44,34 @@ internal object TypeUtils {
         Value(value)
       }
       is Double -> {
+        require(value.isFinite()) { "Value can not be Double.NaN, Double.POSITIVE_INFINITY or Double.NEGATIVE_INFINITY" }
         Value(value)
       }
       is Float -> {
+        require(value.isFinite()) { "Value can not be Float.NaN, Float.POSITIVE_INFINITY or Float.NEGATIVE_INFINITY" }
         Value(value.toDouble())
       }
       is Long -> {
         Value(value)
       }
       is IntArray -> {
-        val valueArray = value.map { Value(it.toLong()) }
+        val valueArray = value.map { wrapToValue(it.toLong()) }
```

Comment:
Why do you need wrap int though?

Reply:
just want to make it consistent, so we can add future assertion to the primitive types as well.